### PR TITLE
chore: simplify installation tests

### PR DESCRIPTION
**Pull request by OrKoN**
This PR changes the installation tests to use mocha context to pass state which allows easily identifying tests and reasoning about the execution steps. It sacrifies the parallel mode because the context cannot be parallelised but that is beneficial because the parallel mode does not support .only, works only for npm and makes it hard to debug tests. I see no difference between the parallel mode and sequential mode in terms of performance.

### CODE REVIEW

File: test/installation/src/puppeteer-webpack.spec.ts [hunk 1]
```diff
@@ -17,42 +17,40 @@
...
 import {join} from 'path';
 
-import {describeInstallation} from './describeInstallation.js';
+import {configureSandbox} from './sandbox.js';
 import {execFile, readAsset} from './util.js';
 
-describeInstallation(
-  '`puppeteer` with Webpack',
-  {
+describe('``puppeteer` with Webpack`', () => {
```
Comment (Lightning00Blade) [inline test/installation/src/puppeteer-webpack.spec.ts]:
```suggestion
describe('`puppeteer` with Webpack', () => {
```

File: test/installation/src/puppeteer.spec.ts [hunk 2]
```diff
@@ -18,30 +18,29 @@ import assert from 'assert';
...
 import {join} from 'path';
 
-import {describeInstallation} from './describeInstallation.js';
+import {configureSandbox} from './sandbox.js';
 import {readAsset} from './util.js';
 
-describeInstallation(
-  '`puppeteer`',
-  {
+describe.only('`puppeteer`', () => {
```
Comment (Lightning00Blade) [inline test/installation/src/puppeteer.spec.ts]:
```suggestion
describe('`puppeteer`', () => {
```

File: test/installation/src/puppeteer-firefox.spec.ts [hunk 3]
```diff
@@ -18,31 +18,30 @@ import assert from 'assert';
...
 import {join} from 'path';
 
-import {describeInstallation} from './describeInstallation.js';
+import {configureSandbox} from './sandbox.js';
 import {readAsset} from './util.js';
 
-describeInstallation(
-  '`puppeteer` with Firefox',
-  {
+describe('``puppeteer` with Webpack`', () => {
```
Comment (Lightning00Blade) [inline test/installation/src/puppeteer-firefox.spec.ts]:
```suggestion
describe('`puppeteer` with Webpack', () => {
```

**Comment by OrKoN** [comment]
Needs more consideration

**Review by Lightning00Blade — APPROVED** [review]

**Review by jrandolf-google — APPROVED** [review]

**Pull request merged**